pep/001-initial-pep-process.md

```diff
@@ -0,0 +1,391 @@
+
+
+#### Specification
+
+The technical specification should describe the command and/or configuration
+syntax and semantics of any new feature.
+
+* If this is a command, we are looking for what the `porter help` would look
+  like: description of command, arguments, flags, default behavior and error
+  handling.
+
+* If this is a syntax change to a configuration file, define the allowed syntax,
+  at least one example per use case, covering defaults and error handling.
+
+* All PEPs will be reviewed for user experience. So make sure to think about the
+  common use case, how people can accomplish more advanced scenarios, precedence
+  from existing Porter features or other tools in the ecosystem, and how the
+  change fits into Porter workflows and tasks.
+
+
+#### Backwards Compatibility
+
+All PEPs that introduce backwards incompatibilities must include a section
+describing these incompatibilities and their severity.  The PEP must explain how
+to deal with these incompatibilities, possibly with defaulting or migrations.
+PEP submissions without a sufficient backwards compatibility treatise may be
+rejected outright.
+
+
+#### Security Implications
+
+If there are security concerns in relation to the PEP, those concerns should be
+explicitly written out to make sure reviewers of the PEP are aware of them.
+
+
+#### Rejected Ideas
+
+Throughout the discussion of a PEP, various ideas will be proposed which are not
+accepted. Those rejected ideas should be recorded along with the reasoning as to
+why they were rejected. This both helps record the thought process behind the
+final version of the PEP as well as preventing people from bringing up the same
+rejected idea again in subsequent discussions.
+
+
+#### Open Questions
+
+While a PEP is in draft, questions can come up which warrant further discussion.
+Those questions should be recorded here so people know that they are being
+thought about but do not have a concrete resolution. This helps make sure all
+for the PEP to be ready for consideration are complete and reduces
+and reduces people duplicating prior discussion. When possible, link the
+question to where it is being discussed, such as a [forum] post/comment.
+
+
+### PEP Lifecycle
+
+We are using the process from Kubernetes Enhancement Proposal for our statuses
+because they work better for our implementation than Python (Porter isn't a spec
+like python, so Kubernetes process fits us better).
+
+This is the happy path, where a PEP is accepted.
+
+1. The idea is posted to the Porter [forum] where the maintainers and community
+   can see it and comment. A maintainer will let you know if it is suitable for
+   a PEP and assign it a number.
+
+1. A pull request for the PEP is submitted with an initial status of
+   `provisional` in the preamble. The status is **PROPOSED** while the pull
+   request is unmerged, and it is not provisionable until the pull request is
+   merged.
+
```

Comment:
is this supposed to be provisional and provisionable? these have the same root but they don't mean the same thing to me?

Reply:
Sorry no that's just my inability to spell.

Reply:
I could totally see it being intentional too.. that's why I was asking.